echosec/torproxy
# SEC-527 Update Tor Proxy and configure for read-only filesystem

## Description of work done
This PR branches Tor Proxy.
- Rebuild the image using a newer version of the base image
- Update config elements to account for updates to the `privoxy` package
- Configures an entrypoint redirect that allows execution in the `/tmp` directory
- Leverages `/tmp` in our Kubernetes deployment as the only read/write directory as part of our security hardening

This fork should be compatible with the license that allows for commercial use and we are making our fork public.
https://github.com/dperson/torproxy/blob/master/LICENSE

## Testing notes
- Pull down the image `echosec.azurecr.io/echosec/torproxy:latest` or alternatively `docker build echosec.azurecr.io/echosec/torproxy:latest .`
- `docker run -dt --rm --name torproxy -p 8118:8118 -p 9050:9050 echosec.azurecr.io/echosec/torproxy:latest`
- Configure Firefox to use the following proxy address `localhost:8118`.

Browse the internet and verify check your IP differs to your standard public IP.
